beetsplug/deezer.py

```diff
@@ -279,12 +279,27 @@ def _search_api(self, query_type, filters=None, keywords=""):
         if not query:
             return None
         self._log.debug(f"Searching {self.data_source} for '{query}'")
-        response = requests.get(
-            self.search_url + query_type,
-            params={"q": query},
-            timeout=10,
-        )
-        response.raise_for_status()
+        try:
+            response = requests.get(
+                self.search_url + query_type,
+                params={"q": query},
+                timeout=10,
+            )
+            response.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            self._log.error(
+                "Error fetching data from {} API\n Error: {}",
+                self.data_source,
+                e,
+            )
+            return None
+        except requests.exceptions.HTTPError as e:
+            self._log.error(
+                "HTTP error fetching data from {} API\n Error: {}",
+                self.data_source,
+                e,
+            )
+            return None
         response_data = response.json().get("data", [])
```

Comment:
Consider handling `response.json()` call as well, see something similar here:

* https://github.com/beetbox/beets/blob/622ed3c7191e20072234748f4584a2f42dc4c460/beetsplug/lyrics.py#L215-L232
* https://github.com/beetbox/beets/blob/622ed3c7191e20072234748f4584a2f42dc4c460/beetsplug/lyrics.py#L812-L814

Reply:
I am trying to understand the need for this. If `data` is missing, `response_data` will be `None`, which seems fine.

Reply:
Sounds all good - some (badly designed) APIs return a 200 OK response with an error as HTML, so if we know that is not the case here it's all good!